homeassistant/components/fronius/sensor.py

```diff
@@ -163,9 +169,40 @@ def available(self):
 
     def entity_description(  # pylint: disable=no-self-use
```

Comment:
Not sure how this is implemented, but `entity_description` is not a method/property.

Generating entity description here doesn't add value IMHO and only adds complexity.

Either define the entity description before creating the sensors entities (create those based on the definitions and pass the description into the entity), or use regular methods.

Reply:
It's much preferred for readability to use a declarative approach and define all of the entity descriptions separately as constants. Then just iterate those or look up the correct description based on the description key.

Reply:
Moved the definition of the properties "last_reset", "device_class" and "state_class" into the sensor class with correct property methods.